### PR TITLE
Create generated-code.md for java

### LINKED ISSUE #508: Create OSS java generated code guide
at `docs/generatedcode/` and link it in docs/index.html

### PR DESCRIPTION
**Pull request by dapengzhang0**
opensourced the internal java codegen guide

resolves #508

### CODE REVIEW

**Comment by dapengzhang0** [comment]
cc @ericgribkoff

File: docs/reference/java/generated-code.md [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+For each service defined in a .proto file, the Java code generation produces a
+Java class. The class name is the service's name suffixed by `Grpc`. The package
+for the generated code is specified in the .proto file using the `java_package`
+option.
+
+For example, if `ServiceName` is defined in a .proto file containing the
+following:
+
+```
```
Comment (mehrdada) [inline docs/reference/java/generated-code.md]: please add `java` after the backtick to this and other multiline code examples in the file to ensure proper highlighting

Comment (dapengzhang0) [inline docs/reference/java/generated-code.md]: done

**Comment by dapengzhang0** [comment]
Per https://support.codebasehq.com/articles/tips-tricks/syntax-highlighting-in-markdown, `protobuf` is supported, and it does highlight. What I'm not sure is `gradle` which was not listed in the document. Is there a way to see the rendered HTML?

**Comment by dapengzhang0** [comment]
I mean can I preview the rendered grpc.io page with my change?

**Comment by mehrdada** [comment]
I think it should work fine. You can build it with Jekyll and try locally, but I'd say let's merge this one and modify if it fails.

**Comment by mehrdada** [comment]
@dapengzhang0 Is Android Java going to have a separate generated code guide? If not, please add a link to the same page under Android Java section as well.

**Comment by dapengzhang0** [comment]
@mehrdada Android codegen is a bit different and was documented here https://grpc.io/docs/tutorials/basic/android.html#generating-client-code
Most android developers are using gradle; Maven is possible but I'm not sure how it looks like, and Bazel codegen rule is not available for the current release yet. So for the time being, the android page itself is precise and sufficient. #543

**Comment by mehrdada** [comment]
I understand—I suggest moving that "Android developers..." paragraph up and implicitly the rest would be non-Android.

**Pull request merged**